File: Jour23/jour23.c
#include<stdio.h>
#include<string.h>
#include<ctype.h>//Contient la fonction isalpha()

#include <stdio.h>
#include <ctype.h>// Contient les fonctions isalpha()/islower()/isupper()

// Fonction pour chiffrer un texte avec une clé donnée
void chiffrerCesar(char* texte, int cle) 
{
	int i;
    for (i = 0; texte[i] != '\0'; i++)
     {
        char c = texte[i];
        if(isalpha(c))
        {
        	// Traitement des majuscules
	        if (isupper(c))
	        {
	            texte[i] = ((c - 'A' + cle) % 26 + 26) % 26 + 'A';
	        }
	        // Traitement des minuscules
	        else if (islower(c)) 
	        {
	            texte[i] = ((c - 'a' + cle) % 26 + 26) % 26 + 'a';
        	}	
        }
        else
        	texte[i] = c;
        
    }
}

int main() {
    char message[100];
    int cle;

    printf("Entrez le message a chiffrer : ");
    fgets(message, sizeof(message), stdin);

    printf("Entrez la cle de decalage : ");
    scanf("%d", &cle);

    chiffrerCesar(message, cle);

    printf("Message transformer : %s\n", message);

    return 0;
}

File: Jour23/jour23.pas
program ChiffrementCesar;
uses crt;

var
  texte: string;
  cle, i: integer;
  caractere: char;

begin
  clrscr;
  write('Entrez le texte a chiffrer : ');
  readln(texte);
  write('Entrez la cle de decalage (0-25) : ');
  readln(cle);
  //Attention aucun control

  for i := 1 to length(texte) do
  begin
        caractere := texte[i];
        if caractere in ['A'..'Z','a'..'z'] then   //Pour verifier si c' est bien une lettre alphabetique
        begin
            // Chiffrement des majuscules
            if (caractere >= 'A') and (caractere <= 'Z') then
            begin
              caractere := chr(((ord(caractere) - ord('A') + cle) mod 26) + ord('A'));
            end;
            // Chiffrement des minuscules
            if (caractere >= 'a') and (caractere <= 'z') then
            begin
              caractere := chr(((ord(caractere) - ord('a') + cle) mod 26) + ord('a'));
            end;
            texte[i] := caractere;
        end 
        else
         texte[i] := caractere;
  end;
writeln('Texte chiffre : ', texte);
end.
